internal/service/alarms/internal/dictionary/dictionary.go

```diff
@@ -0,0 +1,187 @@
+package dictionary
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"sync"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift-kni/oran-o2ims/internal/clients"
+)
+
+const (
+	managedClusterVersionLabel = "openshiftVersion-major-minor"
+	localClusterLabel          = "local-cluster"
+
+	resourceTypeCluster = "Cluster"
+)
+
+type AlarmDictionary struct {
+	Client client.Client
+
+	RulesMap map[int][]*monitoringv1.PrometheusRule
+}
+
+func New(client client.Client) *AlarmDictionary {
+	return &AlarmDictionary{
+		Client:   client,
+		RulesMap: make(map[int][]*monitoringv1.PrometheusRule),
+	}
+}
+
+// Load loads the alarm dictionary
+func (r *AlarmDictionary) Load(ctx context.Context) {
+	// TODO: List Resource Type from Resource Server
+	resourceTypeList := getResourceTypes()
+
+	type result struct {
+		resourceTypeID int
+		rules          []*monitoringv1.PrometheusRule
+		err            error
+	}
+
+	// If there are many resource types, we can use a semaphore to limit the number of concurrent goroutines
+	wg := sync.WaitGroup{}
+	resultChannel := make(chan result)
+	for _, resource := range resourceTypeList {
+		wg.Add(1)
+		go func(resource resourceType) {
+			var err error
+			var rules []*monitoringv1.PrometheusRule
+
+			defer func() {
+				wg.Done()
+				resultChannel <- result{
+					resourceTypeID: resource.id,
+					rules:          rules,
+					err:            err,
+				}
+			}()
+
+			switch resource.model {
+			case resourceTypeCluster:
```

Comment:
This will have to be revisited as the method used to determine if a resource type is a cluster will likely be more complex that just looking for the word "cluster" in the model.

Reply:
Yes, for sure